src/tests/ipc_tests.rs

```diff
@@ -301,17 +309,25 @@ pub mod ipc_tests {
         lindrustfinalize();
     }
 
+    
     pub fn ut_lind_ipc_socketpair() {
+        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently
+        let mut _thelock = TESTMUTEX.lock().unwrap();
+        //creates the file system if it doesnt exist yet.
+        setup::test_setup();
+
         lindrustinit(0);
         let cage = interface::cagetable_getref(1);
-        let mut socketpair = interface::SockPair::default();
+        let socketpairsafe = Arc::new(Mutex::new(interface::SockPair::default()));
```

Comment:
Why does this need to change as part of this?  (I'm curious.  Likely a minor comment about this will suffice...

Reply:
ohh, before we fixed the random ports issue. i was trying a few things to make this stable. but turned out irrelevant. reverting back as this change isnt needed.